paper.md

```diff
@@ -2,11 +2,88 @@
+which does not have the capability to execute many privileged instructions. 
+Therefore, a hypervisor is needed to translate a VM instruction into an instruction 
+that can be executed by the host [@zhang2018comparative]. All these makes using VM's less scalable. Therefore all of these leads us to a question "Can we do better ?".
+
+<!-- what is container tech -->
+
+Although Software (Linux) containers are a relatively old technology [@menage_adding_2007], it was not until recently, with the rise of Docker, that they entered mainstream territory [@bernstein_containers_2014]. Since then, this technology has transformed the way applications get deployed in shared infrastructures, with 25% of companies using this form of software deployment [@datadog_surprising_2018], and a market size projected to be close to 5B by 2023 [@marketsandmarkets_application_2018]. Docker has been the de facto container runtime, with other container runtimes such as Singularity [@kurtzer_singularity_2017], Charliecloud [@priedhorsky_charliecloud_2017] and Podman [@podmancommunity_containers_2019] having emerged. The Linux Foundation bootstrapped the Open Container Initiative (OCI) [@opencontainerinitiative_new_2016] and is close to releasing version 1.0 of a container image and runtime specifications. With Docker, began the era of container-native software development, which is a paradigm that promotes the building, testing and deployment of softwares in containers. Software containers are well-known for being lightweight and
+they solve most of the problems caused by using VM's. So, people started replacing VM's with containers in workflow engines for providing platform independent reproducibility [@repocompscience2016][@piccolo2016tools]. Since, these container runtimes are available for almost every well known operating systems and architectures, experiments can be reproduced easily using containerized workflows in almost any environment [@zheng_integrating_2015].
+
+<!-- this is how people tried solving reproducibility with containers -->
+
+Currently, the different container based workflow execution engines that are available can be classified mainly into two categories. First category involves engines that are inherently container-native but assume the presence of a fully provisioned Kubernetes [@kubernetes_google] cluster at their disposal for workflow execution. Therefore, this category of workflow engines could also be termed as cloud-native [@balalaie2016microservices]. Some popular examples of this type of workflow execution engines are Argo [@argocommunity_argoproj_2019], Pachyderm[@novella_containerbased_2018] and Brigade[@brigade]. The other category of workflow engines do not enforce container based execution by default but provides some plugins and extensions to execute steps in containers. Nextflow [@ditommaso_nextflow_2017] and Pegasus [@deelman_pegasus_2004] are some popular examples of this category.
```

Comment:
we can move the discussion of categorization to the related work section. Also, some of these sentences can be moved to the workflows and virtualization paragraphs. So we can have:
1. problem of reproducibility in science
2. workflows. why they're useful and what's missing
3. containers/kubernetes. why they're useful and what's missing
4. our contributions